src/main/java/httpserver/httpmessages/HTTPRequestHandler.java

```diff
@@ -0,0 +1,17 @@
+package httpserver.httpmessages;
+
+import httpserver.RequestHandler;
+import httpserver.ResponseBuilder;
+
+public class HTTPRequestHandler implements RequestHandler {
+
+    private final ResponseBuilder responseBuilder;
+
+    public HTTPRequestHandler(ResponseBuilder responseBuilder) {
+        this.responseBuilder = responseBuilder;
+    }
+
+    public HTTPResponse handle(HTTPRequest httpRequest) {
+        return responseBuilder.buildResponse(httpRequest.getMethod(), httpRequest.getRequestURI());
```

Comment:
Here you can see that all the `ResponseBuilder` interface gives you is a decomposition of `HTTPRequest` to `(String, String)`